eclair-core/src/main/scala/fr/acinq/eclair/db/sqlite/SqliteAuditDb.scala

```diff
@@ -130,23 +163,23 @@ class SqliteAuditDb(sqlite: Connection) extends AuditDb with Logging {
       statement.executeBatch()
     }
 
-  override def add(e: PaymentRelayed): Unit =
-    using(sqlite.prepareStatement("INSERT INTO relayed VALUES (?, ?, ?, ?, ?, ?)")) { statement =>
-      statement.setLong(1, e.amountIn.toLong)
-      statement.setLong(2, e.amountOut.toLong)
-      statement.setBytes(3, e.paymentHash.toArray)
-      e match {
-        case ChannelPaymentRelayed(_, _, _, fromChannelId, toChannelId, _) =>
-          statement.setBytes(4, fromChannelId.toArray)
-          statement.setBytes(5, toChannelId.toArray)
-        case TrampolinePaymentRelayed(_, _, _, _, fromChannelIds, toChannelIds, _) =>
-          // TODO: @t-bast: we should change the DB schema to allow accurate Trampoline reporting
-          statement.setBytes(4, fromChannelIds.head.toArray)
-          statement.setBytes(5, toChannelIds.head.toArray)
+  override def add(e: PaymentRelayed): Unit = {
+    val payments = e match {
+      case ChannelPaymentRelayed(amountIn, amountOut, _, fromChannelId, toChannelId, ts) => Seq(RelayedPart(fromChannelId, amountIn, "IN", "channel", ts), RelayedPart(toChannelId, amountOut, "OUT", "channel", ts))
+      case TrampolinePaymentRelayed(_, incoming, outgoing, ts) => incoming.map(i => RelayedPart(i.channelId, i.amount, "IN", "trampoline", ts)) ++ outgoing.map(o => RelayedPart(o.channelId, o.amount, "OUT", "trampoline", ts))
+    }
+    for (p <- payments) {
+      using(sqlite.prepareStatement("INSERT INTO relayed VALUES (?, ?, ?, ?, ?, ?)")) { statement =>
```

Comment:
This could be batched? Probably too early to optimize

Reply:
I hesitated, but indeed I thought it was too early :)